internal/util-logging/src/main/scala/sbt/internal/util/Prompt.scala

```diff
@@ -36,4 +37,60 @@ private[sbt] object Prompt {
   private[sbt] case object Watch extends NoPrompt
   private[sbt] case object Pending extends NoPrompt
   private[sbt] case object NoPrompt extends NoPrompt
+  private[sbt] final class Blocked(doJoin: () => Boolean) extends NoPrompt {
```

Comment:
I feel like this construct sticks out since all other `Prompt` states represents a state, and this one is like an underground tunnel between tasks, commands, and virtual IO that's acting like a global variable?
I wonder if some of these blocking imple should move to the individual channel that's doing the blocking?

Reply:
I agree. This aspect is not particularly well designed. The challenge is that a background job could exit it any point so it's kind of tricky to integrate with the sbt main loop. It doesn't really make sense as an sbt state variable because it needs to be managed outside of the logic in main loop. I couldn't think of a great way to handle this problem and this was the best hack that I came up with. I have no doubt there is a better way. I'm just not sure what it is.